### PR TITLE
show warning about CanRequestUserInput instead of skipping

### LINKED ISSUE #44: Known drivers that have no user input is labeled as CanRequestUserInput are skipped
Using `IsInstalled=0` it is somewhat unexpected that `CanRequestUserInput` is skipped. Especially when I know these two updates actually do not request any User Input as these are drivers and never seen any prompts.
```
   {
      "type": "windows-update",
      "search_criteria": "IsInstalled=0"
    },
```

```
vsphere-clone: Skipped (CanRequestUserInput) Windows update (2019-06-06; 0.05 MB): VMware, Inc. - SCSIAdapter - 4/6/2018 12:00:00 AM - 1.3.10.0
vsphere-clone: Skipped (CanRequestUserInput) Windows update (2019-04-24; 0.08 MB): VMware, Inc. - Net - 10/17/2018 12:00:00 AM - 1.8.10.0
```

### PR DESCRIPTION
**Pull request by casz**
it is already possible to provide an exclude filter

This makes CanRequestUserInput optional
fixes #44

If you need this behavior you can use this filter below
```
{
    "provisioners": [
        {
            "type": "windows-update",
            "search_criteria": "IsInstalled=0",
            "filters": [
                "exclude:$_.InstallationBehavior.CanRequestUserInput",
                "include:$true"
            ],
            "update_limit": 25
        }
    ]
}
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,11 @@
 module github.com/rgl/packer-provisioner-windows-update
 
 require (
-	github.com/hashicorp/packer v1.4.0
+	github.com/hashicorp/packer v1.4.5
```
Comment (casz) [inline go.mod]: not sure this is the correct way to fix the dependency issue I saw

```
go: github.com/hashicorp/packer@v1.4.0 requires
        github.com/gophercloud/gophercloud@v0.0.0-20180903124057-ea7289ebdf06: invalid pseudo-version: does not match version-control timestamp (2018-09-17T09:41:35Z)
```

**Comment by casz** [comment]
@rgl I have tested the changes, they work for our use-case.

I did find that it failed to apply cumulative update.

Had to manually boot the machine and run check for updates.

```
==> vsphere-iso: Running Windows update...
    vsphere-iso: Searching for Windows updates...
    vsphere-iso: Found Windows update (2018-10-15; 20.72 MB): Update for Adobe Flash Player for Windows Server 2019 (1809) for x64-based Systems (KB4462930)
    vsphere-iso: Found Windows update (2019-09-10; 20.75 MB): 2019-09 Security Update for Adobe Flash Player for Windows Server 2019 for x64-based Systems (KB4516115)
    vsphere-iso: Found Windows update (2019-10-08; 64.01 MB): 2019-10 Cumulative Update for .NET Framework 3.5, 4.7.2 and 4.8 for Windows Server 2019 for x64 (KB4524099)
    vsphere-iso: Found Windows update (2019-10-15; 63.81 MB): 2019-10 Cumulative Update for .NET Framework 3.5, 4.7.2 and 4.8 for Windows Server 2019 for x64 (KB4520405)
    vsphere-iso: Found Windows update (2019-11-12; 37.02 MB): Windows Malicious Software Removal Tool x64 - November 2019 (KB890830)
    vsphere-iso: Found Windows update (2019-10-19; 0.05 MB): VMware, Inc. - SCSIAdapter - 8/2/2019 12:00:00 AM - 1.3.15.0
    vsphere-iso: Found Windows update (2019-10-12; 0.08 MB): VMware, Inc. - System - 7/11/2019 12:00:00 AM - 9.8.16.0
    vsphere-iso: Found Windows update (2019-10-23; 0.08 MB): VMware, Inc. - Net - 7/11/2019 12:00:00 AM - 1.8.16.0
    vsphere-iso: Found Windows update (2019-11-12; 15014.4 MB): 2019-11 Cumulative Update for Windows Server 2019 (1809) for x64-based Systems (KB4523205)
    vsphere-iso: Downloading Windows updates (9 updates; 15220.91 MB)...
    vsphere-iso: Installing Windows updates...
    vsphere-iso: Waiting for operation to complete (system performance: 51% cpu; 17% memory)...
    vsphere-iso: Waiting for operation to complete (system performance: 50% cpu; 23% memory)...
    vsphere-iso: Waiting for operation to complete (system performance: 55% cpu; 25% memory)...
    vsphere-iso: Waiting for operation to complete (system performance: 6% cpu; 30% memory)...
    vsphere-iso: Waiting for operation to complete (system performance: 46% cpu; 30% memory)...
==> vsphere-iso: Restarting Machine
==> vsphere-iso: Waiting for machine to restart...
==> vsphere-iso: A system shutdown is in progress.(1115)
==> vsphere-iso: #< CLIXML
    vsphere-iso: WIN-SRV-2019 restarted.
==> vsphere-iso: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>
==> vsphere-iso: Machine successfully restarted, moving on
==> vsphere-iso: Uploading the Windows update elevated script...
==> vsphere-iso: Uploading the Windows update check for reboot required elevated script...
==> vsphere-iso: Uploading the Windows update script...
==> vsphere-iso: Running Windows update...
    vsphere-iso: Searching for Windows updates...
    vsphere-iso: Found Windows update (2019-11-12; 15014.4 MB): 2019-11 Cumulative Update for Windows Server 2019 (1809) for x64-based Systems (KB4523205)
    vsphere-iso: Downloading Windows updates (1 updates; 15014.4 MB)...
    vsphere-iso: Installing Windows updates...
```

**Comment by rgl** [comment]
Failing does not seem related to this update; it seems the windows update installs a new network driver update, which might confuse the provisioner (which is not yet reliant enough to survive all errors).

I'm curious, why are you seeing those VMware driver updates? Are you updating an existing image instead of creating it from scratch from an iso (with an up-to-date VMware tools iso)?

**Comment by casz** [comment]
I am creating the image from scratch.
Our VMWare install disc has not been updated within the last year 😰

**Comment by rgl** [comment]
Ah that explains it! I'm using the recent version at https://github.com/rgl/windows-2016-vagrant/blob/ddefee42f3ff3a434c657afbf908e7817aa639f6/windows-2019-vsphere.json#L5 that's why I wan't seeing the same as you.

File: go.mod (hunk 1 above)
Comment (rgl) [reply go.mod]: I've issued the command `go mod tidy` to tidy things up.

**Pull request merged**